batch-add-recording-relationships.user.js

```diff
@@ -38,12 +34,12 @@
 // use or other dealings in this Software without prior written
 // authorization.
 // ==/License==
-
 const scr = document.createElement('script');
-scr.textContent = `(${batch_recording_rels})();`;
+// eslint-disable-next-line prettier/prettier -- cannot prettier-ignore options sepcifically (printWidth: 140); "eslint --fix" splits this big string in a strange way
+scr.textContent = `(${batch_recording_rels})("\\n\\n\\n'''${GM_info.script.name.replace(/"/g, '\\"')}''' ${GM_info.script.version} (relate to '''%edit-mode%''')\\n${GM_info.scriptHandler} ${GM_info.version} ${GM_info.platform ? `(${Object.values(GM_info.platform).join(' ')})` : ''}");`;
```

Comment:
I lost hours trying to pass a variable with this hello of new syntax and that nesting function.

**You tested that working?**

I tried to pass `variable` but maybe not `${variable}` or `${GM_info}`.
I don't understand those $, as I said.

I did all my changes by big blind guessings.
It took more hours.

I also think userscript duly need those wrapper functions but didn't want to include that unrelated change in this merge request.

Reply:
> Unexpected identifier

Well no, I retried but it says that same error I got when trying, on the line after this:

```js
scr.textContent = `(${batch_recording_rels})(${GM_info});`;
```

Reply:
Oops, sorry I haven't tried it myself and there is one thing which I missed: 
The parameter object has to be serialized as JSON, otherwise the injected script would literally  contain `[object Object]` instead of the data itself :sweat_smile:
```js
scr.textContent = `(${batch_recording_rels})(${JSON.stringify(GM_info.script)});`;
```

Reply:
It seems like it should work!

Reply:
Thanks @kellnerd, it is slightly cleaner now.
Even if I still have to exclude some eslint rules for a specific replace I need to do.

Reply:
Thank you. I was going completely nuts with this {`$} syntax I'm not used to (forced by eslint and prettier) and coding on Raspberry Pi\* is super slow as hell, and browser crashes all the time between two tests. 🥴

\* My Linux PC is dead and it seems I cannot install eslint on my office Windows laptop.

Reply:
Yeah, [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) are confusing at first but `` `x${y}z` `` is literally equivalent to a simple string concatenation `'x' + y + 'z'`. The advantage is that the notation is more compact, allows multi-line strings and you don't have to escape quotes.
There is even a fully flexible extension called _tagged_ template strings which allows you to write e.g. ``String.raw`\w` `` for use in regular expressions instead of having to escape the backslash like `'\\w'` (which makes longer regular expressions hard to understand.

Reply:
> My Linux PC is dead and it seems I cannot install eslint on my office Windows laptop.

BTW, Super!
I was working on another repo and found out that it was (without admin rights) **super easy to install `eslint` on the office Windows laptop!**

So, when I have time to go on working on this PR, I can use a faster computer that does not crash all the time! :D